### ISSUE TITLE
Don't create parent Runs if it is not necessary

### ISSUE THREAD

**Issue by Panaetius** [SwissDataScienceCenter/renku-python#1399]

**Comment by rokroskar** [comment]
I agree - this additionally complicates things and results in a different graph depending on whether the step is run for the first time or if it is re-executed from the KG. Ideally running the same thing twice would result in the same graph.
